oit-dual-depth-peeling.html

```diff
@@ -0,0 +1,722 @@
+                }
+
+                gl.bindBuffer(gl.ARRAY_BUFFER, matrixBuffer);
+                gl.bufferSubData(gl.ARRAY_BUFFER, 0, modelMatrixData);
+                
+                gl.drawElementsInstanced(gl.TRIANGLES, sphere.indices.length, gl.UNSIGNED_SHORT, 0, spheres.length);
+
+
+                ////////////////////////////////////
+                // 2. Dual Depth Peeling Ping-Pong
+                ////////////////////////////////////
+                let readId, writeId;
+                let offsetRead, offsetBack;
+
+                for (let pass = 0; pass < NUM_PASS; pass++) {
+                    readId = pass % 2;
+                    writeId = 1 - readId;  // ping-pong: 0 or 1
+                    
+
+                    gl.bindFramebuffer(gl.DRAW_FRAMEBUFFER, allBuffers[writeId]);
+                    gl.drawBuffers([gl.COLOR_ATTACHMENT0]);
+                    gl.clearColor(DEPTH_CLEAR_VALUE, DEPTH_CLEAR_VALUE, 0, 0);
+                    gl.clear(gl.COLOR_BUFFER_BIT);
+
+                    gl.bindFramebuffer(gl.DRAW_FRAMEBUFFER, colorBuffers[writeId]);
+                    gl.drawBuffers([gl.COLOR_ATTACHMENT0, gl.COLOR_ATTACHMENT1]);
+                    gl.clearColor(0, 0, 0, 0);
+                    gl.clear(gl.COLOR_BUFFER_BIT);
+
+                    gl.bindFramebuffer(gl.DRAW_FRAMEBUFFER, allBuffers[writeId]);
+                    gl.drawBuffers([gl.COLOR_ATTACHMENT0, gl.COLOR_ATTACHMENT1, gl.COLOR_ATTACHMENT2]);
+                    gl.blendEquation(gl.MAX);
+
+                    // update texture uniform
+                    offsetRead = readId * 3;
+                    gl.useProgram(dualDepthPeelingProgram);
+                    gl.uniform1i(dualDepthPeelingLocations.depth, offsetRead);
+                    gl.uniform1i(dualDepthPeelingLocations.frontColor, offsetRead + 1);
+
+                    // draw geometry
+                    gl.bindVertexArray(sphereArray);
+                    gl.drawElementsInstanced(gl.TRIANGLES, sphere.indices.length, gl.UNSIGNED_SHORT, 0, spheres.length);
+                    
+                    // blend back color separately
+                    offsetBack = writeId * 3;
+                    gl.bindFramebuffer(gl.DRAW_FRAMEBUFFER, blendBackBuffer);
+                    gl.drawBuffers([gl.COLOR_ATTACHMENT0]);
+                    gl.blendEquation(gl.FUNC_ADD);
+                    gl.blendFuncSeparate(gl.SRC_ALPHA, gl.ONE_MINUS_SRC_ALPHA, gl.ONE, gl.ONE_MINUS_SRC_ALPHA);
+                    gl.useProgram(blendBackProgram);
+                    gl.uniform1i(blendBackLocations.backColor, offsetBack + 2);
+                    gl.bindVertexArray(quadArray);
+                    gl.drawArrays(gl.TRIANGLES, 0, 6);
+                }
+
+                
+
+                //////////////////////////
+                // 3. Final
+                //////////////////////////
+
+                gl.bindFramebuffer(gl.FRAMEBUFFER, null);
+                gl.drawBuffers([gl.BACK]);

```

Comment:
Is this necessary? What does setting it to `gl.BACK` do? I believe `gl.bindFramebuffer(gl.FRAMEBUFFER, null);` automatically makes it so your drawing to the default frame buffer, or am I mistaken?